README.md

```diff
@@ -25,15 +25,27 @@ which can be one of:
 * Wildcard: matches all values.
 * Index: match the value at that index.
 * Text: match all values whose field names are equivalent to that text.
-
+* Annotations: matches values specified by a wrapped path component with the given annotations.
 Some examples:
 ```
-data on reader: {foo: ["foo1", "foo2"] , bar: "myBarValue"}
+data on reader: {foo: ["foo1", "foo2"] , bar: "myBarValue", bar: A::"annotatedValue"}
+
+(foo 0)       - matches "foo1"
+(1)           - matches "myBarValue"
+(*)           - matches ["foo1", "foo2"], "myBarValue" and A::"annotatedValue"
+()            - matches {foo: ["foo1", "foo2"] , bar: "myBarValue", bar: A::"annotatedValue"}
+(foo bar)     - matches "myBarValue" and A::"annotatedValue"
+(foo A::bar)  - matches A::"annotatedValue"
+```
 
-(foo 0) - matches "foo1"
-(1)     - matches "myBarValue"
-(*)     - matches ["foo1", "foo2"] and "myBarValue"
-()      - matches {foo: ["foo1", "foo2"] , bar: "myBarValue"}
+The `()` matcher matches all values in the stream but you can also use annotations with it, example:
+```
+data on reader: 2 3 {} 4 A::2 B::C::[]
+
+()        - matches 2, 3, {}, 4, A::2 and B::C::[]
+A::()     - matches A::2
+B::C::()  - matches B::C::[]
+B::()     - doesn't match anything
```

Comment:
We need an example for a path like `B::(A::foo)` matching the value `42` in the stream `B::{foo: A::42} B::{foo: 43}`, but not the value `43`. Seems like the current implementation doesn't support this.